test/plugin/ckpttimer/ckpttimer.cpp

```diff
@@ -0,0 +1,243 @@
+/* FILE: ckpttimer.cpp
+ * AUTHOR: Rohan Garg
+ * EMAIL: rohgarg@ccs.neu.edu
+ * Copyright (C) 2015 Rohan Garg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <signal.h>
+#include <time.h>
+
+#include "dmtcp.h"
+#include "jassert.h"
+
+#define DEBUG_SIGNATURE "[Ckpttimer Plugin]"
+#ifdef CKPTTIMER_PLUGIN_DEBUG
+# define DPRINTF(fmt, ...) \
+  do { fprintf(stderr, DEBUG_SIGNATURE fmt, ## __VA_ARGS__); } while (0)
+#else
+# define DPRINTF(fmt, ...) \
+  do { } while (0)
+#endif
+
+#define PRINTF(fmt, ...) \
+  do { fprintf(stdout, DEBUG_SIGNATURE fmt, ## __VA_ARGS__); } while (0)
+
+#define handleError(msg) \
+    do { perror(msg); exit(EXIT_FAILURE); } while (0)
+
+
+#define CLOCKID         CLOCK_REALTIME
+#define DEFAULT_SIGN    SIGRTMIN+2
+#define SAMPLE_INTERVAL 1000000000 // 1 second
+#define START_TIMER     true
+#define STOP_TIMER      false
+
+#define PRINT_WARNING           0
+#define PRINT_WARNING_AND_EXIT  1
+
+/* Globals */
+static long long g_interval = SAMPLE_INTERVAL;
+static int g_action  = PRINT_WARNING_AND_EXIT;
+static int g_sig_num = DEFAULT_SIGN;
+
+/* File local functions  */
+static void
+get_and_save_envvars()
+{
+  const char *signal = getenv("DMTCP_CKPTTIMER_SIGNAL");
+  const char *action = getenv("DMTCP_CKPTTIMER_ACTION");
+  const char *interval = getenv("DMTCP_CKPTTIMER_INTERVAL");
+
+#ifdef CKPTTIMER_PLUGIN_TEST
+  static int dummy = 0;
+  while (!dummy);
+#endif
+
+  if (signal) {
+    g_sig_num = atoi(signal);
+    DPRINTF("Using signal (%d) for ckpt timer\n", g_sig_num);
+  } else {
+    g_sig_num = DEFAULT_SIGN;
+  }
+
+  if (action) {
+    g_action = atoi(action);
+  }
+
+  if (interval) {
+    g_interval = atoll(interval);
+  }
+}
+
+static void
+timeout_handler(int sig, siginfo_t *si, void *uc)
+{
+  JWARNING("Checkpoint took longer than expected.");
```

Comment:
Wrong syntax. The format for JWARNING is similar to that of JASSERT.  If you just want to put a note, use JNOTE instead.